File: Problem-035/solution.py
def circular_primes(n):
    count = 0

    # Check every number below n
    for num in range(2, n):
        string = str(num)
        circular = True

        # Check every rotation
        for i in range(len(string)):
            rotated = string[i:] + string[:i]

            if not is_prime(int(rotated)):
                circular = False
                break

        # If all rotations are prime
        if circular:
            count += 1

    return count


def is_prime(m):
    if m < 2:
        return False

    for i in range(2, int(m ** 0.5) + 1):
        if m % i == 0:
            return False

    return True


print(circular_primes(1_000_000))

File: Problem-035/solution.js
function circularPrimes(n) {
    let count = 0;

    for (let num = 2; num < n; num++) {
        let str = num.toString();
        let circular = true;
        for (let i = 0; i < str.length; i++) {
            let rotated = str.slice(i) + str.slice(0, i);

            if (!isPrime(Number(rotated))) {
                circular = false;
                break;
            }
        }
        if (circular) {
            count++;
        }
    }

    return count;
}


function isPrime(m) {
    if (m < 2) return false;

    for (let i = 2; i * i <= m; i++) {
        if (m % i === 0) {
            return false;
        }
    }

    return true;
}

console.log(circularPrimes(1000000));
